modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/impl/DefaultLogStorageFactory.java

```diff
@@ -105,9 +108,18 @@ public DefaultLogStorageFactory(String nodeName, Supplier<Path> logPathSupplier)
         );
     }
 
-    /** {@inheritDoc} */
     @Override
-    public void start() {
+    public CompletableFuture<Void> startAsync() {
+        // This is effectively a sync implementation.
+        try {
+            start();
+            return nullCompletedFuture();
+        } catch (RuntimeException ex) {
+            return failedFuture(ex);
+        }
+    }
+
```

Comment:
By meaning of startAsync -- start() looks like should be started in future: `CompletableFuture.runAsync(() -> start());` The only question is: in most cases a component's executor should be provided.

Reply:
Yes, that's correct. Other implementations also kept a sync implementation after the refactoring.
I intended to write a comment in this line but eventually forgot about it.
Do you think we need to create a new ticket to make this full async in the future? and link it here? Or this is so common that is not necessary.

Reply:
> Do you think we need to create a new ticket to make this full async in the future?

No, I think that's by design

Reply:
So, that's fine then